trace/trace_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trace
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/klog"
+)
+
+func TestStep(t *testing.T) {
+	tests := map[string]*struct {
+		inputString   string
+		expectedTrace *Trace
+	}{
+		"Test 1: When string is empty": {
+			inputString: "",
+			expectedTrace: &Trace{
+				steps: []traceStep{
+					{time.Now(), ""},
+				},
+			},
+		},
+		"Test 2: When string is not empty": {
+			inputString: "test2",
+			expectedTrace: &Trace{
+				steps: []traceStep{
+					{time.Now(), "test2"},
+				},
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			sampleTrace := &Trace{}
+			sampleTrace.Step(tt.inputString)
+			if sampleTrace.steps[len(sampleTrace.steps)-1].msg != tt.expectedTrace.steps[len(tt.expectedTrace.steps)-1].msg {
+				t.Errorf("Failed %s:\n Expected %v \n Got %v \n", name, tt.expectedTrace, sampleTrace)
+			}
+		})
+	}
+}
+
+func TestTotalTime(t *testing.T) {
+	tests := map[string]*struct {
+		inputTrace *Trace
+	}{
+		"Test 1: Test with current system time": {
+			inputTrace: &Trace{
+				startTime: time.Now(),
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := tt.inputTrace.TotalTime()
+			if got == 0 {
+				t.Errorf("Failed %s:\n Got %v \n", name, got)
```

Comment:
The name of the test case doesn't have to be included in the error message. It will be printed as part of `t.Run`. 
```
t.Errorf("expected total time of 0, got %d", got)
```